### PR TITLE
[DOC release] Issue#13678

### LINKED ISSUE #13678: Broken url used inline docs application.js
The link to http://emberjs.com/guides/understanding-ember/the-view-layer/#toc_event-delegation is broken. It can be seen using the github link or documentation link below:

[Location in application.js - https://github.com/emberjs/ember.js/blob/master/packages/ember-application/lib/system/application.js#L149](https://github.com/emberjs/ember.js/blob/master/packages/ember-application/lib/system/application.js#L149)

[Broken link found here in API docs: http://emberjs.com/api/classes/Ember.Application.html](
http://emberjs.com/api/classes/Ember.Application.html)

The line to look for in the documentation is "To learn more about the advantages of event delegation and the Ember view layer, and a list of the event listeners that are setup by default, visit the Ember View Layer guide."

### PR DESCRIPTION
**Pull request by mrosata**
Fixed broken link detailed by issue#13678 according to changes discussed with @pixelhandler 

Fixes #13678

### CODE REVIEW

File: packages/ember-application/lib/system/application.js [hunk 1]
```diff
@@ -144,9 +144,8 @@ export function _resetLegacyAddonWarnings() {
   not receive events.* If you specify a custom root element, make sure you only
   append views inside it!
 
-  To learn more about the advantages of event delegation and the Ember view
-  layer, and a list of the event listeners that are setup by default, visit the
-  [Ember View Layer guide](http://emberjs.com/guides/understanding-ember/the-view-layer/#toc_event-delegation).
+  To learn more about the events Ember components use, see 
+  [components/handling-events](https://guides.emberjs.com/v2.6.0/components/handling-events/#toc_event-names).
```
Comment (rwjblue) [inline packages/ember-application/lib/system/application.js]: We need a good way to reference "current" in these guide links.  

@emberjs/learning-team-managers - Any ideas here?

File: packages/ember-application/lib/system/application.js [hunk 2]
```diff
@@ -144,9 +144,8 @@ export function _resetLegacyAddonWarnings() {
   not receive events.* If you specify a custom root element, make sure you only
   append views inside it!
 
-  To learn more about the advantages of event delegation and the Ember view
-  layer, and a list of the event listeners that are setup by default, visit the
-  [Ember View Layer guide](http://emberjs.com/guides/understanding-ember/the-view-layer/#toc_event-delegation).
+  To learn more about the events Ember components use, see
```
Comment (rwjblue) [inline packages/ember-application/lib/system/application.js]: Looks like there is some trailing whitespace here that is failing the JSCS rules.

Comment (mrosata) [inline packages/ember-application/lib/system/application.js]: Thanks, I'm going to close this PR and start another one with proper formatting.

Comment (pixelhandler) [inline packages/ember-application/lib/system/application.js]: @mrosata would be ok to commit again and then squash before it gets merged.

**Pull request closed without merging**